mobile/src/main/java/mycroft/ai/MainActivity.kt

```diff
@@ -38,466 +38,475 @@ import androidx.appcompat.app.AppCompatActivity
+		utterances.add(mycroftUtterance)
+		binding.contentMain.defaultMessageTextView.visibility = View.GONE
+		mycroftAdapter.notifyItemInserted(utterances.size - 1)
+		if (binding.contentMain.voxswitch.isChecked) {
+			ttsManager.addQueue(mycroftUtterance.utterance)
+		}
+		binding.contentMain.cardList.smoothScrollToPosition(mycroftAdapter.itemCount - 1)
+	}
+
+	private fun registerReceivers() {
+		registerNetworkReceiver()
+		registerWearBroadcastReceiver()
+	}
+
+	private fun registerNetworkReceiver() {
+		if (!isNetworkChangeReceiverRegistered) {
+			// set up the dynamic broadcast receiver for maintaining the socket
+			networkChangeReceiver = NetworkChangeReceiver()
+			networkChangeReceiver.setMainActivityHandler(this)
+
+			// set up the intent filters
+			val connChange = IntentFilter("android.net.conn.CONNECTIVITY_CHANGE")
+			val wifiChange = IntentFilter("android.net.wifi.WIFI_STATE_CHANGED")
+			registerReceiver(networkChangeReceiver, connChange)
+			registerReceiver(networkChangeReceiver, wifiChange)
+
+			isNetworkChangeReceiverRegistered = true
+		}
+	}
+
+	private fun registerWearBroadcastReceiver() {
+		if (!isWearBroadcastRevieverRegistered) {
+			wearBroadcastReceiver = object : BroadcastReceiver() {
+				override fun onReceive(context: Context, intent: Intent) {
+					val message = intent.getStringExtra(MYCROFT_WEAR_REQUEST_MESSAGE)
+					// send to mycroft
+					if (message != null) {
+						Log.d(logTag, "Wear message received: [$message] sending to Mycroft")
+						sendMessage(message)
+					}
+				}
+			}
+
+			LocalBroadcastManager.getInstance(this)
+				.registerReceiver(wearBroadcastReceiver, IntentFilter(MYCROFT_WEAR_REQUEST))
+			isWearBroadcastRevieverRegistered = true
+		}
+	}
+
+	private fun unregisterReceivers() {
+		unregisterBroadcastReceiver(networkChangeReceiver)
+		unregisterBroadcastReceiver(wearBroadcastReceiver)
+
+		isNetworkChangeReceiverRegistered = false
+		isWearBroadcastRevieverRegistered = false
+	}
+
+	private fun unregisterBroadcastReceiver(broadcastReceiver: BroadcastReceiver) {
+		LocalBroadcastManager.getInstance(this).unregisterReceiver(broadcastReceiver)
+	}
+
+	/**
+	 * This method will attach the correct path to the
+	 * [.wsip] hostname to allow for communication
+	 * with a Mycroft instance at that address.
+	 *
+	 *
+	 * If [.wsip] cannot be used as a hostname
+	 * in a [URI] (e.g. because it's null), then
+	 * this method will return null.
+	 *
+	 *
+	 * @return a valid uri, or null
+	 */
+	private fun deriveURI(): URI? {
+		return if (wsip.isNotEmpty()) {
+			try {
+				URI("ws://$wsip:8181/core")
+			} catch (e: URISyntaxException) {
+				Log.e(logTag, "Unable to build URI for websocket", e)
+				null
+			}
+		} else {
+			null
+		}
+	}
+
+	fun sendMessage(msg: String) {
+		// let's keep it simple eh?
+		//final String json = "{\"message_type\":\"recognizer_loop:utterance\", \"context\": null, \"metadata\": {\"utterances\": [\"" + msg + "\"]}}";

```

Comment:
I dislike left over comments. I know not you, but let's clean them up..

Reply:
Boom